include/boost/beast/_experimental/test/handler.hpp

```diff
@@ -35,47 +35,55 @@ class handler
 {
     boost::optional<error_code> ec_;
     bool pass_ = false;
-
+    boost::source_location loc_{BOOST_CURRENT_LOCATION};
 public:
-    handler() = default;
+    handler(boost::source_location loc = BOOST_CURRENT_LOCATION) : loc_(loc) {}
 
     explicit
-    handler(error_code ec)
-        : ec_(ec)
+    handler(error_code ec, boost::source_location loc = BOOST_CURRENT_LOCATION)
+        : ec_(ec), loc_(loc)
     {
     }
 
     explicit
-    handler(boost::none_t)
+    handler(boost::none_t, boost::source_location loc = BOOST_CURRENT_LOCATION) : loc_(loc)
     {
     }
 
-    handler(handler&& other)
+    handler(handler&& other,boost::source_location loc = BOOST_CURRENT_LOCATION)
         : ec_(other.ec_)
         , pass_(boost::exchange(other.pass_, true))
+        , loc_(loc)
+
     {
     }
 
     ~handler()
     {
-        BEAST_EXPECT(pass_);
+        ::boost::beast::unit_test::suite::this_suite()->expect(pass_, loc_.file_name(), loc_.line());
     }
 
     template<class... Args>
     void
     operator()(error_code ec, Args&&...)
     {
-        BEAST_EXPECT(! pass_); // can't call twice
-        BEAST_EXPECTS(! ec_ || ec == *ec_,
-            ec.message());
+        ::boost::beast::unit_test::suite::this_suite()->expect(!pass_, loc_.file_name(), loc_.line());
+        if (ec_ && ec != *ec_)
+            ::boost::beast::unit_test::suite::this_suite()->fail(ec.message(), loc_.file_name(), loc_.line());
+        else
+            ::boost::beast::unit_test::suite::this_suite()->pass();
         pass_ = true;
     }
 
     void
     operator()()
     {
-        BEAST_EXPECT(! pass_); // can't call twice
-        BEAST_EXPECT(! ec_);
+        ::boost::beast::unit_test::suite::this_suite()->expect(!pass_, loc_.file_name(), loc_.line());
+        if (ec_)
```

Comment:
`if( ec_.failed() )`

Reply:
`ec_ ` is the comparison, so if it's set the error-less completion is wrong. but it should be 

`ec_ && ec_->failed()` since ec_ could be set to success (in theory)